diff --git a/riscv/mmu.cc b/riscv/mmu.cc
index d0a55f02..0acde368 100644
--- a/riscv/mmu.cc
+++ b/riscv/mmu.cc
@@ -163,6 +163,14 @@ void mmu_t::load_slow_path(reg_t addr, reg_t len, uint8_t* bytes, uint32_t xlate
 
 void mmu_t::store_slow_path(reg_t addr, reg_t len, const uint8_t* bytes, uint32_t xlate_flags, bool actually_store)
 {
+  
+  if (addr == 0x30000000) {
+    reg_t data = reg_from_bytes(len, bytes);
+    printf("%c", (data & 0xff));
+    fflush(stdout);
+    return;
+  }
+
   reg_t paddr = translate(addr, len, STORE, xlate_flags);
 
   if (!matched_trigger) {
